pkg/addons/addons.go

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package addons
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"k8s.io/minikube/pkg/minikube/assets"
+	"k8s.io/minikube/pkg/minikube/cluster"
+	"k8s.io/minikube/pkg/minikube/command"
+	"k8s.io/minikube/pkg/minikube/config"
+	"k8s.io/minikube/pkg/minikube/exit"
+	"k8s.io/minikube/pkg/minikube/machine"
+	"k8s.io/minikube/pkg/minikube/out"
+	"k8s.io/minikube/pkg/minikube/storageclass"
+	pkgutil "k8s.io/minikube/pkg/util"
+)
+
+// defaultStorageClassProvisioner is the name of the default storage class provisioner
+const defaultStorageClassProvisioner = "standard"
+
+func Set(name, value, profile string) error {
+	a, valid := isAddonValid(name)
+	if !valid {
+		return errors.Errorf("%s is not a valid addon", name)
+	}
+
+	// Run any additional validations for this property
+	if err := run(name, value, profile, a.validations); err != nil {
+		return err
```

Comment:
since there are 6 possible errors emitted by this function, do you mind adding error breadcrumbs to help us when debugging it in the future?

For example:

return errors.Wrap(err, "run")

Reply:
done!